### PR TITLE
Convert OUTPUT_DIRECTORY to optional argument with default value

### LINKED ISSUE #27: Input and output file directories change
# 💡 Summary #

Update the base location of where users will have their input file and output directory.

## Motivation and context ##

This would be useful because currently the input and output, unless a path is specified is inside the repository folder.

## Implementation notes ##

Current example:
tpt-reports RV1234 TEST_ELECTION TEST_DOMAIN TEST_FILE.json TEST_OUTPUT

Right now, TEST_FILE.json needs to be within tpt-reports and TEST_OUTPUT needs will write to tpt-reports.
We'd like it write outside of tpt-reports.

## Acceptance criteria ##

- [x] A user can have the JSON_FILE_PATH outside the repository
- [x] The OUTPUT_DIRECTORY will write outside the repository
- [ ] Default output directory (on instances where argument is not passed) is the root of the users home directory or "~/"

### PR DESCRIPTION
**Pull request by nickviola**
#  #

## 🗣 Description ##
Resolves #27 and converts OUTPUT_DIRECTORY argument to optional "-o --output-dir" with default value decided by team. The default output directory has been set to "~/" (root of users home directory). This PR also patches generate_reports function in test_tpt_reports.py so that it doesn't actually generate pdfs in unit testing.

## 💭 Motivation and context ##
Closes #27, the result of planning discussions with the team to add versatility to the script and simplicity for users.

## 🧪 Testing ##
Run the command line tool without any output directory supplied to use the default "~/" or pass another directory with --output-dir=<SOME_DIRECTORY>.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: src/tpt_reports/tpt_reports.py [hunk 1]
```diff
@@ -1,20 +1,21 @@
 """cisagov/tpt-reports: A tool for creating Technical Phishing Test (TPT) reports.
 
 Usage:
-  tpt-reports ASSESSMENT_ID ELECTION_NAME DOMAIN_TESTED JSON_FILE_PATH OUTPUT_DIRECTORY [--log-level=LEVEL]
+  tpt-reports ASSESSMENT_ID ELECTION_NAME DOMAIN_TESTED JSON_FILE_PATH [--output-dir=OUTPUT_DIRECTORY] [--log-level=LEVEL]
```
Comment (dav3r) [inline src/tpt_reports/tpt_reports.py]: Might as well keep the optional arguments in alphabetical order, same as listed in the "Options" section below:
```suggestion
  tpt-reports ASSESSMENT_ID ELECTION_NAME DOMAIN_TESTED JSON_FILE_PATH [--log-level=LEVEL] [--output-dir=OUTPUT_DIRECTORY]
```

File: src/tpt_reports/tpt_reports.py [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 
 Options:
   -h --help                         Show this message.
   -l --log-level=LEVEL              If specified, then the log level will be set to
                                     the specified value.  Valid values are "debug", "info",
                                     "warning", "error", and "critical". [default: info]
+  --output-dir=OUTPUT_DIRECTORY     The directory where the final PDF reports
```
Comment (dav3r) [inline src/tpt_reports/tpt_reports.py]: Is there a reason why you didn't include a short flag here, e.g. `-o` or `-d`?  Note that if you go with `-d`, you'll want to update the options to keep them in alphabetical order by short flag.

File: tests/test_tpt_reports.py [hunk 3]
```diff
@@ -170,3 +175,64 @@ def test_parse_json():
...
+        "test",
+        TEST_JSON_FILE,
+    ]
+
+    # Patch usage arguments
+    with patch.object(sys, "argv", patched_args):
+        # Set mock return value
+        mock_generate_reports.return_value = True
+
+        # Call tpt_reports.main entry poifunction
```
Comment (dav3r) [inline tests/test_tpt_reports.py]: I think this is what you meant?
```suggestion
        # Call tpt_reports.main entry point function
```

**Review by dav3r — COMMENTED** [review]
I have a few small notes for your consideration.  Also, please add some additional details to the PR description so that it mentions all of changes that are included in this branch.

File: src/tpt_reports/tpt_reports.py (hunk 2 above)
Comment (nickviola) [reply src/tpt_reports/tpt_reports.py]: I had the "-o" flag there initially but decided I would get opinions from the team first and forgot to add it back in. I think it makes sense to add "-o" for consistency here. I'll get it updated. Thanks!

File: tests/test_tpt_reports.py (hunk 3 above)
Comment (nickviola) [reply tests/test_tpt_reports.py]: Yes. 😂 My brain stopped processing here apparently. Thanks!

File: src/tpt_reports/tpt_reports.py (hunk 2 above)
Comment (nickviola) [reply src/tpt_reports/tpt_reports.py]: Resolved in a610b15

Comment (dav3r) [reply src/tpt_reports/tpt_reports.py]: Looks great! 👍

**Review by dav3r — APPROVED** [review]
👍 👍

**Review by schmelz21 — APPROVED** [review]

**Comment by nickviola** [comment]
Since this PR has already been approved, I will put the versioning changes in my next PR per discussions here: https://github.com/cisagov/tpt-reports/pull/51#issuecomment-1737566932

**Pull request merged**